main.js

```diff
@@ -0,0 +1,97 @@
+let searchBtn = document.querySelector(".search");
+let weatherDisplay = document.querySelector(".todays-weather");
+let forecastCards = document.querySelectorAll(".forecastDisplay .col-md");
+
+// I had to rely on code copilot to help me format how the code is organized below. I couldnt get things to work properly alone. 
+// I mostly struggled on the fetch API and then manipulating/formating the data that was fetched. But I understand how it works
+
+searchBtn.addEventListener("click", function () {
+    let userInput = document.querySelector(".city-input").value.trim();
+    document.querySelector(".city-input").value = ""; 
+
+    if (!userInput) {
+        alert("Please enter a city name!");
+        return;
+    }
+
+    const fetchCordinates = (userInput) => {
+        const url = `http://api.openweathermap.org/geo/1.0/direct?q=${userInput}&limit=1&appid=5fcdd42d3867910bc60aa5df701c10d9`;
+        return fetch(url, { method: 'GET' })
+            .then(response => response.json())
+            .then(data => {
+                return {
+                    Lat: data[0].lat,
+                    Lon: data[0].lon,
+                };
+            });
+    };
+
+    fetchCordinates(userInput)
+        .then(cordinates => {
+            getCurrWeather(cordinates);
```

Comment:
you are calling this getCurrentWeather but this function returns nothing. Better calling it fetchWeather. Also you are displaying the weather inside of this function. This is confusing and prone to bugs. Be clear with what the functions are doing